extensions/Skyhigh173/bigint.js

```diff
@@ -0,0 +1,456 @@
+(function(Scratch){
+  'use strict';
+
+  const bi = x => {
+    try {
+      if (x.slice(-1) == 'n') {
+        return BigInt(x.slice(0, -1));
+      }
+      return BigInt(x);
+    } catch {
+      try {
+        return BigInt(Math.trunc(Number(x)));
+      } catch {
+        return 0n;
+      }
+    }
+  };
+
+  const makeLabel = (text) => ({
+    blockType: 'label',
+    text: text
+  });
+
+
+  class BigIntExtension {
+    getInfo() {
+      return {
+        id: 'skyhigh173BigInt',
+        name: 'BigInt',
+        color1: '#59C093',
+        blocks: [
+          {
+            opcode: 'bigint_from',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'To BigInt [text]',
+            arguments: {
+              text: {
+                type: Scratch.ArgumentType.STRING,
+                defaultValue: ''
+              }
+            }
+          },
+          {
+            opcode: 'bigint_as',
```

Comment:
from and as should probably just be merged

also, it's worth thinking about what "6.5 to bigint" should do. cutting off the decimals seems like a reasonable solution

Reply:
turns out the other blocks already do truncate decimals, just these don't